src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,109 @@
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">5</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                laboriosam mollitia et enim quasi adipisci quia provident illum
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">6</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                qui ullam ratione quibusdam voluptatem quia omnis
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
-
-          <tr data-cy="todo">
-            <td className="is-vcentered">7</td>
-            <td className="is-vcentered"> </td>
-
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">
-                illo expedita consequatur quia in
-              </p>
-            </td>
-
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
-                <span className="icon">
-                  <i className="far fa-eye" />
-                </span>
-              </button>
-            </td>
-          </tr>
+  const todos = useAppSelector(state => state.todos);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const handleButton = useCallback(
+    (todo: Todo) => {
+      dispatch(setCurrentTodo(todo));
+    },
+    [dispatch],
+  );
 
-          <tr data-cy="todo">
-            <td className="is-vcentered">8</td>
+  const visibleTodos = useMemo(() => {
+    let filteredTodos;
+
+    switch (status) {
+      case 'all':
+        filteredTodos = todos;
+        break;
+      case 'active':
+        filteredTodos = todos.filter(todo => todo.completed === false);
+        break;
+      case 'completed':
+        filteredTodos = todos.filter(todo => todo.completed === true);
+        break;
+      default:
+        return todos;
```

Comment:
In the default case of the switch statement, return `filteredTodos` instead of `todos` to ensure any subsequent filtering logic is applied consistently.